lib/internal/url.js

```diff
@@ -932,6 +935,60 @@ function urlToOptions(url) {
   return options;
 }
 
+function getPathFromURLWin32(url) {
+  var hostname = url.hostname;
+  var pathname = url.pathname;
+  if (hostname !== '') {
+    // If hostname is set, then we have a UNC path
```

Comment:
… where decoding of percent escapes is handled by the OS? If yes, it might be nice to have the comment extended by that. :)